meshio/abaqus/_abaqus.py

```diff
@@ -245,17 +247,17 @@ def _read_set(f, params_map):
     set_ids = []
     while True:
         line = f.readline()
-        if line.startswith("*"):
+        if not line or line.startswith("*"):
             break
         if line.strip() == "":
             continue
 
         set_ids += [int(k) for k in line.strip().strip(",").split(",")]
 
-    if "generate" in params_map:
+    if "GENERATE" in params_map:
```

Comment:
Is this always capitalized? If not, we could check for either or the two. Don't know the abaqus spec by heart.

Reply:
You are right, it does not have to be capitalized. So I added `upper()` when parsing the arguments.